### ISSUE TITLE
Save Entities modal shows 'Untitled' for terms with updated term meta

### ISSUE THREAD

**Issue by bvlgn** [WordPress/gutenberg#54350]
### Description

When the user updates post related term meta the related terms are labeled 'Untitled' in the Save dialog
This should show the name of the updated term(s).

### Step-by-step reproduction instructions

1. Install and activate the attached issue test/demo plugin on a test site:
[Term Meta Save Entities Bug Demo Plugin.zip](https://github.com/WordPress/gutenberg/files/12575939/Term.Meta.Save.Entities.Bug.Demo.Plugin.zip)
 _The plugin basically only registers a test term meta field and adds a inspector panel with a text control for the test term meta field._

2. Edit a post
3. Make sure a category is assigned to the post 
4. Open the test panel added by the test plugin:

![panel](https://github.com/WordPress/gutenberg/assets/13752843/807697bc-24a9-4658-a2ec-7156f786cae1)

5. Edit the text in the input from 'Hello ' to e.g. 'Hello everybody'
6. Click on the Update button 
7. Have a look at the modal and see that the label of the term is 'Untitled' instead of the name of the term:
 
![modal](https://github.com/WordPress/gutenberg/assets/13752843/1ec151d2-e13d-4466-b715-19086ba15ffd)

### Environment info

WP 6.3.1
GB 16.6.0 (same problem when not active)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by bvlgn** [comment]
Re-tested this with Gutenberg 16.8.1 and the issue is still relevant.

**Comment by yashjawale** [comment]
## Reproduction Report
### Description
This report validates whether the issue can be reproduced.

### Environment
- WordPress: 6.9-alpha-60272
- PHP: 8.2.28
- Server: Apache/2.4.62 (Debian)
- Database: mysqli (Server: 11.4.7-MariaDB-ubu2404 / Client: mysqlnd 8.2.28)
- Browser: Firefox 140.0
- OS: macOS
- Theme: Twenty Twenty-Five 1.2
- MU Plugins: _[preinstalled by Gutenberg's dev environment]_
  * Gutenberg Test Plugin, Disable Login Autofocus 
  * Gutenberg Test Plugin, Disable Remote Patterns 
  * Gutenberg Test Plugin, Enable Templates UI 
  * Gutenberg Test Plugin, No-cache Headers 
  * Gutenberg Test Plugin, Normalize Theme 
  * server-timing.php 
- Plugins:
  * Gutenberg 21.2.0
  * Gutenberg Test Term Meta Field 1.0 - _Test plugin provided by issue author_
  * Test Reports 1.2.0

### Actual Results
1.  ✅ Modal shows name of taxonomy as `Untitled` instead of actual taxonomy name. Issue reproduced.

### Supplemental Artifacts

https://github.com/user-attachments/assets/0d9ed18f-5d62-4043-b980-9788fb55b665

**Comment by SirLouen** [comment]
== Reproduction Report
=== Description
✅ This report validates that the issue can be reproduced.

=== Environment
- WordPress: 6.9-alpha-60093-src
- PHP: 8.2.29
- Server: Apache/2.4.62 (Debian)
- Database: mysqli (Server: 11.8.2-MariaDB-ubu2404 / Client: mysqlnd 8.2.29)
- Browser: Chrome 138.0.0.0
- OS: Windows 10/11
- Theme: Twenty Fourteen 4.3
- MU Plugins: None activated
- Plugins:
  * Gutenberg Test Term Meta Field 1.0
  * Test Reports 1.2.0

=== Testing Instructions
- We can use the OP instructions for testing,
- But I used a slightly modified version to test this with slug instead of meta terms (check additional notes)

=== Actual Results
1.  ✅ Error condition occurs (reproduced).

=== Additional Notes
- I have added a variation to the plugin provided by @bvlgn, instead of creating and modifying a new metafield, what I'm doing is modifying the slug (just to modify something not in meta table). And the problem is still there.

- So basically, we can conclude that modifying Terms from the Editor and saving, as suggested, on the Saved Entities Dialog, the modified Taxonomy name is shown as Untitled instead of the actual taxonomy name

- So now we double bug reproduction, I think we can remove `Needs Testing` and this is ready for `Needs Dev` 

cc @jordesign @Mamaduka 

=== Supplemental Artifacts

<img width="525" height="319" alt="Image" src="https://github.com/user-attachments/assets/616f2641-941d-4de2-ace0-d49ff683bf99" />

![](https://i.imgur.com/sNoOVf7.png)

<img width="471" height="380" alt="Image" src="https://github.com/user-attachments/assets/7fac1242-ba95-4124-bf77-5d448d3911fc" />

**Comment by Mamaduka** [comment]
Here's the fix - https://github.com/WordPress/gutenberg/pull/70863.
